readme.md

```diff
@@ -71,9 +77,11 @@ Returns the first filepath found (by respecting the order) or `null`.
 
 #### filename
 
-Type: `string`
+Type: `string` `Function`
+
+Filename of the file to find, or a custom matcher function to be called with each directory until it returns a filepath to stop the search or the root directory has been reached and nothing was found. When using a matcher function, if you want to check whether a file exists, use [`fs.access()`](https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback) - this is done automatically when `filename` is a string.
```

Comment:
I think it would be useful for us to at least expose convenience methods to check whether the path exists in a normalized way. `fs.access()` is not obvious to use and you would have to manually promisify it when using it async.

We already depend on [`path-exists`](https://github.com/sindresorhus/path-exists) in `locate-path`, so we could depend on it and expose it here somehow without any increase in dependencies. I think this makes sense as checking whether the path exist is the most common operation you would do.

Reply:
I think my documentation made it sound worse than it is. `fs.access()` isn't really necessary. In most use cases for `find-up`, it's sufficient to use `fs.existsSync()` if the filesystem even needs to be checked, and that returns a simple boolean, making it much easier to use here. I've added an example of that to the Usage section instead. Hopefully, a little simpler/clearer now.

I did consider that maybe we should ignore the result of the matcher if the filepath it returns does not exist. But I think it's highly likely that the matcher will be checking the filesystem anyway. And by leaving this up to the matcher, we also enable the user to decide where the search stops in a more deterministic manner (i.e. its decision will never be overridden). I think this is a good foundation to build on top of.

As for convenience methods, are you thinking something like `findUp.exists(matcher)` / `findUp.existsSync(matcher)`? Or maybe a pipeline like `findUp(matcher, findUp.exists)`? My immediate reaction is this merely saves a few characters. If we were to add this, I'd prefer the latter pipeline pattern.